src/scripts/eastern_kingdoms/eastern_plaguelands/naxxramas/boss_patchwerk.cpp

```diff
@@ -111,76 +108,71 @@ struct boss_patchwerkAI : public ScriptedAI
             m_pInstance->SetData(TYPE_PATCHWERK, IN_PROGRESS);
             
         m_events.ScheduleEvent(EVENT_BERSERK, BERSERK_TIMER);
-        m_events.ScheduleEvent(EVENT_HATEFULSTRIKE, HATEFUL_CD);
+        m_events.ScheduleEvent(EVENT_HATEFULSTRIKE, urand(1500, 2000));
 #if SUPPORTED_CLIENT_BUILD >= CLIENT_BUILD_1_12_1
         m_events.ScheduleEvent(EVENT_SLIMEBOLT, SLIMEBOLT_INITIAL);
 #endif
     }
 
     void DoHatefulStrike()
     {
-        // The ability is used on highest HP target in melee
-        // current tank cannot be hit by hateful as long as there are other players in melee
-        
-        // todo: can it hit anything other than players?
-
-        Unit* mainTank = m_creature->GetVictim();
-        
-        // Shouldnt really be possible, but hey, weirder things have happened
-        if (!mainTank)
-            return;
-        ObjectGuid const& mainTankGuid = mainTank->GetObjectGuid();
+        // Hateful Strike sources:
+        // https://classic.wowhead.com/guides/patchwerk-naxxramas-raid-strategy
+        // https://old.reddit.com/r/classicwow/comments/jw3vlt/patchwerk_hateful_strike_clarification/
 
-        Unit* pTarget = nullptr;
-        uint32 uiHighestHP = 0;
+        Unit* pTarget            = nullptr;
+        uint32 uiHighestHP       = 0;
         uint8 threatListPosition = 0;
 
         ThreatList const& tList = m_creature->GetThreatManager().getThreatList();
         for (const auto iter : tList)
         {
-            // Only top 4 players on threat in melee range are targetted.
+            // Do not check further than fourth form threat list
             if (threatListPosition > 3)
                 break;
 
-            if (!iter->getUnitGuid().IsPlayer())
-                continue;
-            
-            Player* pTempTarget = m_creature->GetMap()->GetPlayer(iter->getUnitGuid());
-            if (!pTempTarget)
-                continue;
-
-            if (!m_creature->IsInMap(pTempTarget))
-                continue;
-
-            if (!m_creature->CanReachWithMeleeSpellAttack(pTempTarget))
-                continue;
-
-            // Skipping maintank, only using him if there is no other viable target 
-            // todo: not sure if this is correct. Should we target the MT over the offtanks, if the offtanks have less hp?
-            if (iter->getUnitGuid() != mainTankGuid)
+            // Skip current main tank for inital target search
+            if (threatListPosition > 0)
             {
-                // target has higher hp than anyone checked so far
-                if (pTempTarget->GetHealth() > uiHighestHP)
+                // Only players
+                if (iter->getUnitGuid().IsPlayer())
```

Comment:
Can pets on the threat list be targeted?

Reply:
I've just added this to the remaing TODO's. 
(With this change Hateful Strike can only hit a pet, if pet is main tank and no player from threat list place 2,3,4 is within meele range)